tabular/tests/unittests/test_tabular.py

```diff
@@ -90,7 +90,11 @@ def _assert_predict_dict_identical_to_predict(predictor: TabularPredictor, data)
                 else:
                     model_pred = predictor.predict(data, model=m, as_pandas=as_pandas)
                 if as_pandas:
-                    assert model_pred.equals(predict_dict[m])
+                    if model_pred.dtype == "object":
+                        assert model_pred.equals(predict_dict[m])
+                    else:
+                        # pandas default int type on Windows is int64, while on Linux it is int32
+                        assert model_pred.astype("int64").equals(predict_dict[m].astype("int64"))
```

Comment:
This is incorrect when the prediction is a float, such as 0.7, for regression.

Instead:

```
if model_pred.dtype in ["int64", "int32"]:
    assert predict_dict[m].dtype in ["int64", "int32"]
    assert model_pred.astype("int64").equals(predict_dict[m].astype("int64"))
else:
    assert model_pred.equals(predict_dict[m])
```